server/app/services/question/types/QuestionType.java

```diff
@@ -56,6 +56,14 @@ public boolean isMultiOptionType() {
   }
 
   public static QuestionType of(String name) throws InvalidQuestionTypeException {
+    // Match label, e.g. "Phone Number" -> PHONE
+    for (QuestionType type : QuestionType.values()) {
```

Comment:
this function still feels weird to me because it's trying two different things, and why things are different isn't really documented. Maybe you could just factor this portion out to a fromLabel function and leave the existing function as is.